server_test.go

```diff
@@ -33,32 +33,36 @@ import (
 	. "gopkg.in/check.v1"
 )
 
-// API suite container.
-type TestSuiteFS struct {
+// API suite container common to both FS and XL.
+type TestSuiteCommon struct {
+	serverType string
 	testServer TestServer
 	endPoint   string
 	accessKey  string
 	secretKey  string
 }
 
-// Initializing the test suite.
-var _ = Suite(&TestSuiteFS{})
+// Init and run test on FS backend.
+var _ = Suite(&TestSuiteCommon{serverType: "FS"})
+
+// Init and run test on XL backend.
+var _ = Suite(&TestSuiteCommon{serverType: "XL"})
```

Comment:
This guarantees running of the tests twice right ? one for "XL" and one for "FS" ?